config/modules/acme/default.nix

```
@@ -1,19 +1,40 @@
-{ config, ... }:
+{ config, lib, pkgs, ... }:
+
+with lib;
+
 {
-  users.groups.acme = { };
+  options = {
+    security.acme.nameserver = mkOption {
+      type = types.str;
+      description = ''
+        Nameserver to push RFC2136 updates to.
+      '';
+    };
+  };
+
+  config = {
+    users.groups.acme = { };
 
-  security.acme = {
-    email = "mail@hackint.org";
-    acceptTerms = true;
-    certs = {
-      "${config.networking.fqdn}" = {
-        extraDomainNames = [
-          "hackint.org"
-          "irc.hackint.org"
-        ];
-        dnsProvider = "rfc2136";
-        credentialsFile = /etc/nixos/secrets/acme/environment;
-        group = "acme";
+    security.acme = {
+      email = "mail@hackint.org";
+      acceptTerms = true;
+      certs = {
+        "${config.networking.fqdn}" = {
+          extraDomainNames = [
+            "hackint.org"
+            "irc.hackint.org"
+          ];
+          dnsProvider = "rfc2136";
+          credentialsFile = pkgs.writeText "acme-credentials" ''
+            LEGO_EXPERIMENTAL_CNAME_SUPPORT=true
+            RFC2136_NAMESERVER=${config.security.acme.nameserver}
+            RFC2136_TSIG_ALGORITHM=hmac-sha512.
+            RFC2136_TSIG_KEY=update
+            RFC2136_TSIG_SECRET_FILE=/etc/nixos/secrets/acme/update.tsig
+            RFC2136_PROPAGATION_TIMEOUT=3600
+          '';
```

```suggestion
            RFC2136_SEQUENCE_INTERVAL=60
          '';
```
it's another default I talked about recently on mumble. It implies that between each subject altname there is a 60 seconds pause.

We don't have to modify it, but I think it would be prudent to at least explicitly add it here, so someone looking for a minute of timeout can find it.

---

Added